server/src/metadata/collection.js

```diff
@@ -38,10 +29,23 @@ class Collection {
   }
 
   set_table(table) {
-    table.collection = this;
-    this._table = table;
-    this._waiters.forEach((done) => this._table.on_ready(done));
-    this._waiters = [ ];
+    if (table) {
+      table.collection = this;
+      this.table = table.table;
+      this._table = table;
+      this._waiters.forEach((done) => this._table.on_ready(done));
+      this._waiters = [ ];
+    } else {
+      const old_table = this._table;
+      this.table = null;
+      this._table = null;
+
+      if (old_table) {
```

Comment:
It seems like if we need to do cleanup when removing a table here, we need to do it in the branch above as well.